src/weekly/export.js

```diff
@@ -78,26 +78,28 @@ async function processSubs (downloadStream: ReadStream, deliveryDate: moment, st
   let csvStream = csv.parse({
     headers: true
   })
-      .on('data-invalid', function (data) {
-        // TODO CAN WE LOG PII?
-        console.log('ignoring invalid data: ' + data)
+    .on('data-invalid', function (data) {
+      // TODO CAN WE LOG PII?
+      console.log('ignoring invalid data: ' + data)
+    })
+    .on('data', (data) => {
+      let subscriptionName = data[SUBSCRIPTION_NAME]
+      let changeType = data['RatePlan.AmendmentType']
+      let endDate = moment(data['RatePlanCharge.EffectiveEndDate'])
+      if (changeType === 'RemoveProduct' && deliveryDate.isSameOrAfter(endDate)) return
```

Comment:
why can't you add this condition to the Zuora query filter?

Reply:
Because it doesn't negate easily and ZOQL doesn't support NOT clauses.

Reply:
can't you do something like :
( RatePlan.AmendmentType != 'RemoveProduct' OR  deliveryDate < RatePlanCharge.EffectiveEndDate )

Reply:
Thanks, I spent a while on this on Friday and didn't come to !(A&&B) == (!A || !B)